From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MC_XiaoHei <xor7xiaohei@gmail.com>
Date: Sat, 2 Aug 2025 09:46:11 +0800
Subject: [PATCH] Leaves Plugin

This patch is licensed under the MIT license.

diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 4a606e0eb142aa66721f4a065ce33babdd50ffa9..57936d3924395f455506bd0326b6cab25fda6bb7 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -3010,6 +3010,12 @@ public final class Bukkit {
         server.restart();
     }
 
+    // Leaves start - Feature API
+    public static org.leavesmc.leaves.plugin.FeatureManager getFeatureManager() {
+        return server.getFeatureManager();
+    }
+    // Leaves end - Feature API
+
     // Leaves start - Bot API
     /**
      * Returns a bot manager.
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index eae8087657ac93136af55ba257555db049da55fe..4e548f3c14a043619c14a9afb9365314da3e0b5b 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2775,6 +2775,10 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     void allowPausing(@NotNull org.bukkit.plugin.Plugin plugin, boolean value);
     // Paper end - API to check if the server is sleeping
 
+    // Leaves start - Feature API
+    org.leavesmc.leaves.plugin.FeatureManager getFeatureManager();
+    // Leaves End - Feature API
+
     // Leaves start - Bot API
     /**
      * Returns a bot manager.
