From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 06:52:46 +0000
Subject: [PATCH] Dont send useless entity packets


diff --git a/net/minecraft/server/level/ServerEntity.java b/net/minecraft/server/level/ServerEntity.java
index 60af82925d2f7e605bf7340cf01f1a5668f4d725..a0a38800f740f8c7b241c50e3c176559ade6263c 100644
--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -199,6 +199,11 @@ public class ServerEntity {
                     sentPosition = true;
                     sentRotation = true;
                 }
+                // Leaves start - dont send useless entity packets
+                if (packet != null && org.leavesmc.leaves.LeavesConfig.performance.dontSendUselessEntityPackets && isUselessPacket(packet)) {
+                    packet = null;
+                }
+                // Leaves end - dont send useless entity packets
 
                 if (this.entity.needsSync || this.trackDelta || this.entity instanceof LivingEntity livingEntity && livingEntity.isFallFlying()) {
                     Vec3 movement = this.entity.getDeltaMovement();
@@ -305,6 +310,21 @@ public class ServerEntity {
         this.positionCodec.setBase(this.entity.position());
     }
 
+    // Leaves start - dont send useless entity packets
+    private boolean isUselessPacket(Packet<?> possibleUselessPacket) {
+        if (possibleUselessPacket instanceof ClientboundMoveEntityPacket packet) {
+            if (possibleUselessPacket instanceof ClientboundMoveEntityPacket.Pos) {
+                return packet.getXa() == 0 && packet.getYa() == 0 && packet.getZa() == 0;
+            } else if (possibleUselessPacket instanceof ClientboundMoveEntityPacket.PosRot) {
+                return packet.getXa() == 0 && packet.getYa() == 0 && packet.getZa() == 0 && packet.getYRot() == 0 && packet.getXRot() == 0;
+            } else if (possibleUselessPacket instanceof ClientboundMoveEntityPacket.Rot) {
+                return packet.getYRot() == 0 && packet.getXRot() == 0;
+            }
+        }
+        return false;
+    }
+    // Leaves end - dont send useless entity packets
+
     public void removePairing(final ServerPlayer player) {
         this.entity.stopSeenByPlayer(player);
         player.connection.send(new ClientboundRemoveEntitiesPacket(this.entity.getId()));
