From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 06:54:49 +0000
Subject: [PATCH] Config to disable method profiler


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 11a28268165b6a1e2b2c92c193b4873de14bafa0..a4aad552262955355654d9c6a4570ef5eda89a8f 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1347,7 +1347,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
                 // Paper - improve tick loop - done above
 
-                try (Profiler.Scope ignored = Profiler.use(this.createProfiler())) {
+                try (Profiler.Scope ignored = org.leavesmc.leaves.LeavesConfig.mics.disableMethodProfiler ? null : Profiler.use(this.createProfiler())) { // Leaves - disable profiler
                     this.processPacketsAndTick(sprinting);
                     ProfilerFiller profiler = Profiler.get();
                     profiler.push("nextTickWait");
diff --git a/net/minecraft/util/profiling/Profiler.java b/net/minecraft/util/profiling/Profiler.java
index ee9794784a53cbe4ec02ec4ad94923ce8381b5dd..2f6f5075a04d9f2562e75f0099e72ce0acb7ce72 100644
--- a/net/minecraft/util/profiling/Profiler.java
+++ b/net/minecraft/util/profiling/Profiler.java
@@ -45,6 +45,7 @@ public final class Profiler {
     }
 
     public static ProfilerFiller get() {
+        if (org.leavesmc.leaves.LeavesConfig.mics.disableMethodProfiler) return InactiveProfiler.INSTANCE; // Leaves - disable profiler
         return ACTIVE_COUNT.get() == 0 ? getDefaultFiller() : Objects.requireNonNullElseGet(ACTIVE.get(), Profiler::getDefaultFiller);
     }
 
