From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 06:54:50 +0000
Subject: [PATCH] Remove lambda from ticking guard


diff --git a/net/minecraft/server/level/ServerLevel.java b/net/minecraft/server/level/ServerLevel.java
index 177437b877eecf5aa895c5e7c675235abdf19559..2c58172a9f6f9b3b988a1a6b7a278c8e71f4aaf0 100644
--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -903,7 +903,24 @@ public class ServerLevel extends Level implements WorldGenLevel, ServerEntityGet
                                     }
 
                                     profiler.push("tick");
-                                    this.guardEntityTick(this::tickNonPassenger, entity);
+                                    // Leaves start - copied from this.guardEntityTick
+                                    if (org.leavesmc.leaves.LeavesConfig.performance.remove.tickGuardLambda) {
+                                        try {
+                                            this.tickNonPassenger(entity); // Leaves - changed
+                                        } catch (Throwable throwable) {
+                                            if (throwable instanceof ThreadDeath) throw throwable; // Paper
+                                            // Paper start - Prevent block entity and entity crashes
+                                            final String msg = String.format("Entity threw exception at %s:%s,%s,%s", entity.level().getWorld().getName(), entity.getX(), entity.getY(), entity.getZ());
+                                            MinecraftServer.LOGGER.error(msg, throwable);
+                                            getCraftServer().getPluginManager().callEvent(new com.destroystokyo.paper.event.server.ServerExceptionEvent(new com.destroystokyo.paper.exception.ServerInternalException(msg, throwable))); // Paper - ServerExceptionEvent
+                                            entity.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DISCARD);
+                                            // Paper end - Prevent block entity and entity crashes
+                                        }
+                                        this.moonrise$midTickTasks(); // Paper - rewrite chunk system
+                                    } else {
+                                        this.guardEntityTick(this::tickNonPassenger, entity);
+                                    }
+                                    // Leaves end - copied from this.guardEntityTick
                                     profiler.pop();
                                 }
                             }
