From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 07:00:00 +0000
Subject: [PATCH] Old hopper suckin behavior


diff --git a/net/minecraft/world/level/block/entity/HopperBlockEntity.java b/net/minecraft/world/level/block/entity/HopperBlockEntity.java
index c8fab5a630c9b981dd190df7d77830d43af9a6f5..73dffaed88364bea1f4e996742f0f66ee7b9c1a6 100644
--- a/net/minecraft/world/level/block/entity/HopperBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/HopperBlockEntity.java
@@ -568,8 +568,8 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
             return false;
         } else {
             boolean isBlocked = hopper.isGridAligned()
-                && blockState.isCollisionShapeFullBlock(level, blockPos)
-                && !blockState.is(BlockTags.DOES_NOT_BLOCK_HOPPERS);
+                && (!org.leavesmc.leaves.LeavesConfig.modify.oldMC.oldHopperSuckInBehavior && blockState.isCollisionShapeFullBlock(level, blockPos))
+                && !blockState.is(BlockTags.DOES_NOT_BLOCK_HOPPERS); // Leaves - oldHopperSuckInBehavior
             if (!isBlocked) {
                 for (ItemEntity entity : getItemsAtAndAbove(level, hopper)) {
                     if (addItem(hopper, entity)) {
