From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 07:00:00 +0000
Subject: [PATCH] Bytebuf API


diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index b2b57824894ffdee994b0360edb71d6269b53484..e7d807225ded38b10703952cb3a8fb895c58f130 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -295,6 +295,13 @@ public abstract class PlayerList {
         boolean immediateRespawn = gameRules.get(GameRules.IMMEDIATE_RESPAWN);
         boolean reducedDebugInfo = gameRules.get(GameRules.REDUCED_DEBUG_INFO);
         boolean doLimitedCrafting = gameRules.get(GameRules.LIMITED_CRAFTING);
+
+            // Leaves start - Bytebuf API
+            if (!(player instanceof org.leavesmc.leaves.bot.ServerBot) && !(player instanceof org.leavesmc.leaves.replay.ServerPhotographer)) {
+                org.leavesmc.leaves.bytebuf.internal.InternalBytebufHandler.updatePlayer(player);
+            }
+            // Leaves end - Bytebuf API
+
         playerConnection.send(
             new ClientboundLoginPacket(
                 player.getId(),
