From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 07:01:53 +0000
Subject: [PATCH] Old zombie reinforcement


diff --git a/net/minecraft/world/entity/monster/zombie/Zombie.java b/net/minecraft/world/entity/monster/zombie/Zombie.java
index aa0a34e3bae3c0d81bc87577570340d1efc46c2b..d5ed4718268d7e07ca91a5807cd1967997037e88 100644
--- a/net/minecraft/world/entity/monster/zombie/Zombie.java
+++ b/net/minecraft/world/entity/monster/zombie/Zombie.java
@@ -333,7 +333,7 @@ public class Zombie extends Monster {
             int x = Mth.floor(this.getX());
             int y = Mth.floor(this.getY());
             int z = Mth.floor(this.getZ());
-            EntityType<? extends Zombie> type = this.getType();
+                EntityType<? extends Zombie> type = org.leavesmc.leaves.LeavesConfig.modify.oldMC.oldZombieReinforcement ? EntityType.ZOMBIE : this.getType(); // Leaves end - old zombie reinforcement
             Zombie reinforcement = type.create(level, EntitySpawnReason.REINFORCEMENT);
             if (reinforcement == null) {
                 return true;
