From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 07:02:30 +0000
Subject: [PATCH] Prevent loss of item drops due to update suppression when


diff --git a/net/minecraft/server/level/ServerPlayerGameMode.java b/net/minecraft/server/level/ServerPlayerGameMode.java
index adba2fef82fd7186436165b8be6293b1479a466b..563980ceb15b817e44460fb19c3bd624dda5ae5c 100644
--- a/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -407,11 +407,21 @@ public class ServerPlayerGameMode {
         this.level.captureDrops = new java.util.ArrayList<>();
         // CraftBukkit end
         BlockState adjustedState = org.leavesmc.leaves.command.leaves.subcommands.BlockUpdateCommand.isNoBlockUpdate() ? state : block.playerWillDestroy(this.level, pos, state, this.player); // Leaves - no block update
-        boolean changed = this.level.removeBlock(pos, false);
+                // Leaves start - Prevent loss of item drops due to update suppression when breaking blocks
+                boolean changed;
         if (SharedConstants.DEBUG_BLOCK_BREAK) {
             LOGGER.info("server broke {} {} -> {}", pos, adjustedState, this.level.getBlockState(pos));
         }
-
+                org.leavesmc.leaves.util.UpdateSuppressionException ex = null;
+                try {
+                    changed = this.level.removeBlock(pos, false);
+                } catch (org.leavesmc.leaves.util.UpdateSuppressionException e) {
+                    ex = e;
+                    ex.provideBlock(level, pos, block);
+                    ex.providePlayer(this.player);
+                    changed = false;
+                }
+                // Leaves end - Prevent loss of item drops due to update suppression when breaking blocks
         if (changed) {
             block.destroy(this.level, pos, adjustedState);
         }
@@ -435,6 +445,7 @@ public class ServerPlayerGameMode {
         if (event.isDropItems()) {
             org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockDropItemEvent(bblock, bState, this.player, itemsToDrop); // Paper - capture all item additions to the world
         }
+                if (ex != null) throw ex; // Leaves - Prevent loss of item drops due to update suppression when breaking blocks
 
         // Drop event experience
         if (changed) {
diff --git a/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java b/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
index d1ce11fbb3fbf330b2005435d29c30b9a6e85c9e..332f0053867f75d9f8d3f43ac50c6b6eaabb5e13 100644
--- a/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
@@ -150,6 +150,16 @@ public abstract class AbstractFurnaceBlockEntity extends BaseContainerBlockEntit
     }
     // CraftBukkit end
 
+    // Leaves start - pca
+    @Override
+    public void setChanged() {
+        super.setChanged();
+        if (org.leavesmc.leaves.LeavesConfig.protocol.pca.enable) {
+            org.leavesmc.leaves.protocol.PcaSyncProtocol.syncBlockEntityToClient(this);
+        }
+    }
+    // Leaves end - pca
+
     @Override
     protected void loadAdditional(final ValueInput input) {
         super.loadAdditional(input);
