From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LeavesMC Test CI <ci-test@leavesmc.org>
Date: Fri, 28 Aug 2026 07:07:15 +0000
Subject: [PATCH] Leaves Extra Yggdrasil Service


diff --git a/src/main/java/com/destroystokyo/paper/profile/PaperAuthenticationService.java b/src/main/java/com/destroystokyo/paper/profile/PaperAuthenticationService.java
index 52a8786082712bb735fdc768a228169829fb814b..0b0e779640c92cbf8657a6c5d1200f759ea0bca2 100644
--- a/src/main/java/com/destroystokyo/paper/profile/PaperAuthenticationService.java
+++ b/src/main/java/com/destroystokyo/paper/profile/PaperAuthenticationService.java
@@ -11,7 +11,7 @@ import java.net.Proxy;
 
 public class PaperAuthenticationService extends YggdrasilAuthenticationService {
 
-    private final Environment environment;
+    protected final Environment environment; // Leaves - private -> protected
 
     public PaperAuthenticationService(Proxy proxy) {
         super(proxy);
